packages/tinacms-clerk/src/tinacms.ts

```diff
@@ -0,0 +1,74 @@
+import { AbstractAuthProvider } from 'tinacms'
+
+import type Clerk from '@clerk/clerk-js'
+
+export class ClerkAuthProvider extends AbstractAuthProvider {
+  clerk: Clerk
+  allowedList?: string[]
+  constructor({
+    clerk,
+    allowedList,
+  }: {
+    clerk: Clerk
+    /**
+     * For premium Clerk users, you can use restrictions
+     * https://clerk.com/docs/authentication/allowlist
+     */
+    allowedList?: string[]
+  }) {
+    super()
+    this.clerk = clerk
+
+    this.allowedList = allowedList
+  }
+  /**
+   * Generates a short-lived token when Tina makes a request
+   */
+  async getToken() {
+    await this.clerk.load()
+    if (this.clerk.session) {
+      return { id_token: await this.clerk.session.getToken() }
+    }
+  }
+
+  async logout() {
+    await this.clerk.load()
+    await this.clerk.session.remove()
+  }
+  async authenticate() {
+    this.clerk.openSignIn({
+      redirectUrl: '/admin/index.html', // This should be the Tina admin path
+      appearance: {
+        elements: {
+          // Tina's sign in modal is in the way without this
+          modalBackdrop: { zIndex: 20000 },
+          // Some styles clash with Tina's styling
+          socialButtonsBlockButton: 'px-4 py-2 border border-gray-100',
+          formFieldInput: `px-4 py-2`,
+          formButtonPrimary: 'bg-blue-600 text-white p-4',
+          formFieldInputShowPasswordButton: 'm-2',
+          dividerText: 'px-2',
+        },
+      },
+    })
+  }
+  async getUser() {
```

Comment:
instead of doing this, we should implement authorize like here: https://github.com/tinacms/tina-cloud-starter-self-hosted-clerk/blob/kldavis4/clerk_users_collection/tina/config.tsx#L17

That way, you aren't immediately logged out and get a helpful not authorized message

Reply:
Updated